app/src/main/java/com/example/ithaca_transit_android_v2/MainActivity.kt

```diff
@@ -1,50 +1,134 @@
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.activity_toolbar_search.*
+import kotlinx.android.synthetic.main.item_searchview.*
 
+class MainActivity : AppCompatActivity(), OnMapReadyCallback {
+    private lateinit var disposable: Disposable
+    private var mSearchLocations: List<Location> = ArrayList()
+    private var mSearchAdapter: SearchViewAdapter? = null
 
-class MainActivity : AppCompatActivity() {
+    override fun onMapReady(map: GoogleMap?) {
+        map!!.setOnMapClickListener { point ->
+            Log.i("qwerty","map clicked")
+            search_view.clearFocus() }
+    }
+
+    private fun createSearchObservable(): Observable<SearchState> {
+        val obs = Observable.create { emitter: ObservableEmitter<SearchState> ->
+            val watcher: TextWatcher = object : TextWatcher {
+                override fun afterTextChanged(p0: Editable?) {
+                }
+
+                override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                }
+
+                override fun onTextChanged(searchText: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                    if (searchText!!.length == 0) {
+                        emitter.onNext(EmptyInitClickState())
+                    } else {
+                        emitter.onNext(InitSearchState(searchText.toString()))
+                    }
+                }
+            }
+
+            search_input.addTextChangedListener(watcher)
+
+            search_input.setOnFocusChangeListener { view, hasFocus ->
+                if (hasFocus && (view as EditText).text.isEmpty()) {
+                    emitter.onNext(EmptyInitClickState())
+                } else if (hasFocus) {
+                    emitter.onNext(InitSearchState((view as EditText).text.toString()))
+                }
+                else {
+                    emitter.onNext(SearchLaunchState())
+                }
+            }
+
+            emitter.setCancellable {
+                search_input.removeTextChangedListener(watcher)
+                search_input.setOnFocusChangeListener(null)
+            }
+        }
+        return obs.startWith(SearchLaunchState())
+    }
+
+    fun initSearchView() {
+        val observable = createSearchObservable()
+
+        disposable = observable
+            .observeOn(Schedulers.io())
+            .map { state ->
+                if (state is InitSearchState) {
+                    val locations = NetworkUtils().getSearchedLocations(state.searchText)
+                    InitLocationsSearchState(state.searchText, locations)
+
+                } else {
+                    state
+                }
+            }
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe ({ state ->
+                when (state) {
+                    is SearchLaunchState -> {
```

Comment:
technically this should be implemented in a view delegate class 
(ie. `state -> SearchViewDelegate.render(state)` where `fun render(SearchState state) { state -> when ...}`)
but honestly, if that unnecessarily complicates the app let's just move this to the presenter class

Reply:
Ok. We can talk about this during work session